src/utils/ObjectUtils.ts

```diff
@@ -107,10 +90,24 @@ export function shallowEqual(obj1: any, obj2: any): Boolean {
       return false;
     }
     if (obj1.hasOwnProperty(property) && obj2.hasOwnProperty(property)) {
-      if (obj1[property] !== obj2[property]) {
-        return false;
+      if (shallow) {
+        if (obj1[property] !== obj2[property]) {
+          return false;
+        }
+      } else {
+        if (!_equal(obj1[property], obj2[property], false)) {
```

Comment:
Nit: could just return this.

Reply:
:nod:

Reply:
quick update, ya can't actually! since this is iterating over every property, returning the value here will short circuit the logic and could return a false-positive if this logic isn't the last property checked